src/routes/session.ts

```diff
@@ -1,29 +1,79 @@
+import * as _ from 'lodash';
 import {
 	loginUserXHR,
 	findUser,
 	getUser,
 	comparePassword,
 } from '../platform/auth';
-import { sbvrUtils } from '../platform';
+import { User as DbUser } from '../models';
+import { resinApi, root, sbvrUtils } from '../platform';
 import { captureException, handleHttpErrors } from '../platform/errors';
 import { resetCounter } from '../lib/rate-limiting';
 import { RequestHandler } from 'express';
 import { SetupOptions } from '..';
 
 const { BadRequestError, NotFoundError } = sbvrUtils;
 
-export const whoami: RequestHandler = (req, res) =>
-	getUser(req)
-		.then(user =>
-			res.send({ id: user.id, username: user.username, email: user.email }),
-		)
-		.catch(err => {
-			if (handleHttpErrors(req, res, err)) {
-				return;
+export const whoami: RequestHandler = async (req, res) => {
+	try {
+		const user = await getUser(req);
+
+		// If the user email is available, then the record was successfully
+		// populated by the passport middleware
+		if (user.email !== undefined) {
+			return res.send({
+				id: user.id,
+				username: user.username,
+				email: user.email,
+			});
+		} else {
+			// If we get to this point, then the request is most likely being made
+			// using a scoped access token.
+			// First we check to see if this token can access the user resource,
+			// then if it can, we retrieve user document that corresponds to the
+			// `actor` value on the token using root privileges.
+			const [userWithId] = (await resinApi.get({
+				resource: 'user',
+				passthrough: { req },
+				options: {
+					$filter: {
+						actor: user.actor,
```

Comment:
I'm pretty sure `user.actor` can be undefined so imo we should check that it's not before running this query